### DISCUSSION TITLE
Why does V2 require React 18?

### DISCUSSION THREAD

**Discussion by evan-buss** [KevinVandy/material-react-table#814] [General]
Hello,

I am currently unable to upgrade my project to React 18 and will be stuck on React 17 for the immediate future. After trying to install V2, I learned that MRT now requires React 18.

Can you give any insight as to why this is the case? As far as I can tell, neither [MUI](https://github.com/mui/material-ui/blob/b62d5757e6dc218c57cf6db77dea78c7b2396a5a/packages/mui-material/package.json#L94-L95) or [@tanstack/react-table](https://github.com/TanStack/table/blob/a1e9732e6fc3446a2ae80db72a1f2b46a5c11e46/packages/react-table/package.json#L48-L50) depend upon React 18.

Thanks for the great library, it's incredibly useful.

> Side note: the search functionality for the V1 docs seems to have been recently shut off 😞

**Comment by KevinVandy** [comment]
I was originally experimenting with some react 18 and concurrent mode features to speed up render performance during filtering. I actually haven't put any react 18 dependent features in MRT v2 yet, but plan to. Also since it's almost 2024 and 2 years since React 18 was released, I thought it was pretty expected to recommend React 18.

You can upgrade React 18 but keep your app rendering in React 17 mode btw.

**Reply by evan-buss** [reply]
Unfortunately the project is stuck on 17 due to using the Microsoft SharePoint [SPFX framework](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/compatibility#spfx-development-environment-compatibility) which has its own non-standard build toolchain. They are incredibly slow to update versions.

**Comment by KevinVandy** [comment]
And yeah I removed search from the v1 docs because Docs search is only set up for my main domain name. The v1 docs are legacy, and just there for a while until people migrate.
